### DISCUSSION TITLE
SMTP setup with Gmail Workspace

### DISCUSSION THREAD

**Discussion by lxbanov** [zitadel/zitadel#8846] [Q&A] [answered]
A company of ours owns a Google Workspace with dedicated email addresses for the employees. We wish to setup **self-hosted** Zitadel to send emails using the address `auth@malevich.ai`. I am trying to setup SMTP provider using my private [Google app password ](https://support.google.com/accounts/answer/185833?hl=en) and Google Workspace Account username.

I have followed the steps:
1. Enabled 2FA at my company Google Account.
2. Generated a 16-digit app password.
3. Configured SMTP relay at Google Admin Panel:
![image](https://github.com/user-attachments/assets/787ec261-22dc-41dc-9ae0-4f8a95b98111)
4. Configured SMTP Provider at Zitadel instance with following parameters:
![image](https://github.com/user-attachments/assets/f51c2a8e-f563-44fb-b19d-a592a79845f4)
![image](https://github.com/user-attachments/assets/4fc3e7c2-98a7-4056-bae1-b7788d1d00b8)

When I am trying to send email at the step 3 in SMTP provider configuration, I get the following error:

```
RpcError(INTERNAL): could not add smtp auth, check if both your user and password are correct, if they're correct maybe your provider requires an auth method not supported by ZITADEL (EMAIL-s9kfs)
```

The only logs that appear are:
```
time="2024-10-30T12:12:03Z" level=info msg=activity caller="/home/runner/work/zitadel/zitadel/internal/activity/activity.go:105" domain="https://auth.malevich.ai" grpcStatus=13 httpStatus=500 instance=291460791041786940 isSystemUser=false method=/zitadel.admin.v1.AdminService/TestSMTPConfig org=291460791041852476 path= requestMethod= trigger=resourceAPI user=291460791042507836
```

I tried to use the same credentials and setup at [smtp test service](https://www.gmass.co/smtp-test) and it successfully sent letters using most of the combinations of Port/Security settings.

I wonder if anyone could help us to setup SMTP provider correctly or debug the issue.

**Comment by prashansa-shukla** [comment]
Hi @lxbanov, facing the same issue. were you able to solve it?

**Comment by fforootd** [comment]
Hm, have you tried enabling TLS on the Zitadel config?

It looks like in Google this is enabled.

**Comment by CallumBoase** [comment]
Also having this issue. Did anyone fix it?

I can successfully send email using the same credentials from a test SMTP service, but it doesn't work in Zitadel
I'm trying to use the smpt relay service for gmail (g-suite paid plan)

Here is my setup (i've shown placeholders in the screenshots but have it filled with my correct details)
![image](https://github.com/user-attachments/assets/110436cf-4664-453c-add1-2edb08ddd01c)
![image](https://github.com/user-attachments/assets/f85561a9-c518-4eb5-80a3-3ee49bf6827e)

Then when testing
![image](https://github.com/user-attachments/assets/df0c9cb6-32e3-4c8b-94c1-4e4fafa1615a)

**Comment by CallumBoase** [comment]
Oh, I was able to solve it by using SMTP instead of the SMTP relay service.

So instead of the host and port being smtp-relay.gmail.com:587, use smtp.gmail.com:587

Gmail recommends the relay service, so it would be great if Zitadel supported that rather than reverting to the non-recommended direct smtp.gmail.com method

**Comment by AliaksandrTsimokhau** [comment]
Hi @fforootd , do you already have plans or a timeline for implementing full OAuth2 support for generic SMTP provider which should probably solve the issue with connecting to smtp-relay.gmail.com?

**Reply by fforootd** [reply]
Hm I need to check... the other day we talked about this topic in the context of office 365.

@elinashoko might know what's up there.

**Comment by elinashoko** [comment, marked as answer]
Hey @AliaksandrTsimokhau we've planned to add Oauth2 support for SMTP and will be working on it shortly, you can subscribe here: https://github.com/zitadel/zitadel/issues/8042
However, it's worth calling out out that Google’s SMTP relay service (smtp-relay.gmail.com) does not use OAuth2 at all. With Gmail Relay, you don’t log in with a user account. Google allows email only from servers or domains you’ve approved in advance in your Workspace admin settings.

Currently that's not supported in Zitadel. I haven't heard many community requests so far, however if that's something you need - feel free to open a feature request in our [public repo](https://github.com/zitadel/zitadel) and we can gather feedback from the rest of the community and prioritise accordingly.

**Reply by AliaksandrTsimokhau** [reply]
Thank you for the input and insight! I appreciate the clarification on Google's SMTP relay service and the distinction with OAuth2. I'll subscribe to #8042 to stay updated on the OAuth2 support for SMTP. Looking forward to the implementation!
